src/util.rs

```diff
@@ -297,3 +299,31 @@ macro_rules! create_and_verify_proof {
 pub(crate) fn get_power_of_two_iter<F: ff::PrimeField>() -> impl Iterator<Item = F> {
     iter::successors(Some(F::ONE), |l| Some(l.double()))
 }
+
+pub(crate) struct CellsValuesView<'l, F: PrimeField> {
```

Comment:
Is the main purpose of this struct to debug?

Reply:
Yes, to write important nuances to logs and debug info without a huge amount of garbage.